pkg/common/chunker.go

```diff
@@ -0,0 +1,35 @@
+package common
+
+import (
+	"bufio"
+	"errors"
+	"io"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	ChunkSize = 10 * 1024
+	PeekSize  = 3 * 1024
+)
+
+func ChunkReader(r io.Reader) chan []byte {
+	chunkChan := make(chan []byte)
+	go func() {
+		defer close(chunkChan)
+		reader := bufio.NewReaderSize(bufio.NewReader(r), ChunkSize)
+		for {
+			chunk := make([]byte, ChunkSize)
+			n, err := reader.Read(chunk)
+			if err != nil && !errors.Is(err, io.EOF) {
+				log.WithError(err).Error("Error chunking reader.")
+			}
```

Comment:
**Question:** Should we `continue` here?

Reply:
Good catch. I think we should `break` to avoid getting stuck in the loop.